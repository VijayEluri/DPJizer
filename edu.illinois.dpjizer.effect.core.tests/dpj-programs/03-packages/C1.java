/**
 * This file is licensed under the University of Illinois/NCSA Open Source License. See LICENSE.TXT for details.
 */
package packages;

/**
 * 
 * @author Mohsen Vakilian
 * @author Danny Dig
 *
 */
class C1 {
	region r1;
	int x in r1;
	
	C1() {}
	
	void m1() {
		x = 3;
		C2 c2 = new C2();
		c2.m2();
	}
}

class C2 {
	region r2;
	int y in r2;
	
	C2() {}
	
	void m2() {
		if (y == 3) {
			C1 c1 = new C1();
			c1.m1();
		}
	}
}