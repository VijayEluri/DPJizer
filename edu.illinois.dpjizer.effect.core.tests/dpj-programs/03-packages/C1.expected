packages.C1.C1() invokes java.lang.Object.Object()
packages.C1.m1() writes Root : packages.C1.r1 invokes packages.C2.m2() where { this -> Root : * }
packages.C2.C2() invokes java.lang.Object.Object()
packages.C2.m2() reads Root : packages.C2.r2 invokes packages.C1.m1() where { this -> Root : * }